ssg/oval_object_model/oval_document.py

```diff
@@ -107,7 +107,9 @@ def product_name(self, __value):
     @staticmethod
     def _skip_if_is_none(value, component_id):
         if value is None:
-            raise MissingOVALComponent(component_id)
+            #raise MissingOVALComponent(component_id)
+            logging.warning("Component {} is None".format(component_id))
+            return True
```

Comment:
I'm not sure what was the initial idea of this function but in its current state it just hard-coded to raise an exception. For me it seems reasonable to throw a warning instead. If you @jan-cerny or @Honny1 have a better idea — I'm open to suggestions.

Reply:
This exception is that there should be no references to an OVAL component that does not appear in the OVAL document. For example, a missing state or test object.  

The same warning is issued in the [`validate_references`](https://github.com/ComplianceAsCode/content/blob/2bd80b67657e487332bce5ac70c1fbe23e891661/ssg/oval_object_model/oval_document.py#L134) function. Other cases where this expectation may be raised are when references are collected in the OVAL Definition. It should be up to the programmer to handle this case after calling the [`get_all_references_of_definition`](https://github.com/ComplianceAsCode/content/blob/2bd80b67657e487332bce5ac70c1fbe23e891661/ssg/oval_object_model/oval_container.py#L213) function.

I recommend that you keep `raise MissingOVALComponent(component_id)`.

Reply:
The problem with the exception is that it stands in the way when the Reference object is used to actually determine what is missing and what is orphaned.

Reply:
E.g. `_ensure_by_xccdf_referenced_oval_no_extra_def_in_oval_file` from this PR.

Reply:
To get a list of missing references. I would suggest creating a method of OVAL Document that provides the missing references and bypasses the exception invocation.

Reply:
Okay, I've played with this a bit more, and I think we could be able to just live with the Exeption after fixing the last part that gives us this problem (https://github.com/ComplianceAsCode/content/pull/11432). This, after all, is good mechanism that prevents content authors from doing nasty things (see the PR).